### DISCUSSION TITLE
Anisotropic thermal expansion

### DISCUSSION THREAD

**Discussion by avtarsinghh1991** [idaholab/moose#19931] [Q&A Modules: General] [answered]
Hello MOOSE experts

I am solving a coupled chemo-mechanical problem for an anisotropic material. I am using the finite strain tensor mechanics module.  Can you please let me know the procedure to add the anisotropic thermal expansion? Please suggest any related examples as well. Thank you very much.

**Material section in the input file is as follows:**

```
[Materials]
  [./elasticity_tensor_with_Euler]
    type = ComputeElasticityTensorCP
    fill_method = symmetric9
    C_ijkl = '1.984e9 0.196e9 0.176e9 1.684e9 0.176e9 1.584e9 0.754e9 0.754e9 0.754e9'
    read_prop_user_object = euler_angle_read
  [../]

 [thermalexpansion]
    type = ComputeThermalExpansionEigenstrain
    temperature = diffused
    thermal_expansion_coeff = 3.496e-6
    stress_free_temperature = 0.0
    eigenstrain_name = thermal_strain
  []

  [stress]
    type = ComputeFiniteStrainElasticStress
  []
[]
```

**Comment by ngrilli** [comment, marked as answer]
Dear @asingh-mit 

I am not sure if there is ready code in moose for that.
The way I do is by introducing the thermal eigenstrain in the stress calculation, see example here:
https://github.com/ngrilli/c_pfor_am/blob/main/src/materials/FiniteStrainCrystalPlasticityThermal.C

I modify the function calcResidual by introducing thermal_eigenstrain
in my case it is isotropic but you can make it anisotropic.

Beware that if you express it in the lattice reference frame, then you will need to transform it using _crysrot
before using because the calculations in calcResidual are done in the global reference frame.

Best Regards,
Nicolò

**Reply by avtarsinghh1991** [reply]
Hello Nicolo

Thank you very much. Actually, I sorted this out using "ComputeVariableEigenstrain" material module.
https://mooseframework.inl.gov/source/materials/ComputeVariableEigenstrain.html

ϵ=pT

with p defines the prefactor which I assumed as (c-c_ref) and T is the rank 2 tensor which I used as an anisotropic type.

**In input file**

  [./eigen_strain_prefactor]
    type = DerivativeParsedMaterial
    args = c
    f_name = eigen_strain_prefactor
    constant_names = 'c_ref'
    constant_expressions = '0.0'
    function = (c-c_ref)/3.0
  [../]

  [./eigenstrain]
    type = ComputeVariableEigenstrain
    args = c
    eigen_base = '3.496e-6 0 0 0 3.496e-6 0 0 0 3.496e-6'
    prefactor = eigen_strain_prefactor
    eigenstrain_name = eigenstrain
  [../]


Please let me know if I am doing anything wrong. I shall be very thankful.

Best
Avtar
